db/lnbolt/peerdb.go

```diff
@@ -55,7 +55,13 @@ func (pdb *peerboltdb) GetPeerAddrs() ([]lncore.LnAddr, error) {
 			if k == nil {
 				break
 			}
-			atmp = append(atmp, lncore.LnAddr(string(k)))
+			lnaddr, err := lncore.ParseLnAddr(string(k))
+			if err != nil {
+				logging.Warnf("lnbolt/peerdb: found invalid key in DB as lnaddr: %s (error: %s)", string(k), err.Error())
```

Comment:
shouldn't we ideally check while writing and not reading? one should be enough if we're doing it right.

Reply:
Benefits outweigh the risks, if "magic" happens and the DB gets corrupted then it would be good to have an error about it here before it caused problems somewhere else where the original source would have been unknown.

Reply:
a partial corruption where the key exists but the value doesn't? seems a bit weird, but yeah, doesn't hurt to have one I guess.